src/renderer/gl_engine/tvgGlRenderer.cpp

```diff
@@ -1041,6 +1102,26 @@ bool GlRenderer::render(TVG_UNUSED RenderCompositor* cmp, const RenderEffect* ef
         gaussianTask->vertTask->setDrawRange(ioffset, 6);
         // add task to render pipeline
         pass->addRenderTask(gaussianTask);
+    } // effect fill 
+    else if ((effect->type == SceneEffect::Fill) || (effect->type == SceneEffect::Tint) || (effect->type == SceneEffect::Tritone)) {
+        GlProgram* program{};
+        if (effect->type == SceneEffect::Fill) program = mPrograms[RT_EffectFill];
+        else if (effect->type == SceneEffect::Tint) program = mPrograms[RT_EffectTint];
+        else if (effect->type == SceneEffect::Tritone) program = mPrograms[RT_EffectTritone];
```

Comment:
maybe like that? to avoid checking twice for the effect type?
```   
    } else {
        GlProgram* program = nullptr;

        if (effect->type == SceneEffect::Fill) program = mPrograms[RT_EffectFill];
        else if (effect->type == SceneEffect::Tint) program = mPrograms[RT_EffectTint];
        else if (effect->type == SceneEffect::Tritone) program = mPrograms[RT_EffectTritone];
        else return;
```